### DISCUSSION TITLE
Symbols are taken as plain text ("at" instead of @)

### DISCUSSION THREAD

**Discussion by Rumeysakeskin** [openai/whisper#2320] [Q&A] [answered]
When I try to provide an email id, symbols like @ are printed as plain text.

For example the audio contains is My email is data@test. com 
After using whisper the out text is showing My email is data at the right test. com

How can I solve this issue so that it can show symbols instead of plain text in output?
I also want a multilingual solution for all languages.

**Comment by misutoneko** [comment, marked as answer]
Hi, you might be able to it with just by postprocessing the result (via scripting, or whatever...).
This means that you would be looking for signs of a domain name that comes after the word "at".
The complication is that nowadays you can have a lot of variety in domain names, it's not plain old dot com.

Other than that, prompting/finetuning. There are old posts about those.
I've used prompting myself with varying success, it's worth trying out imho (no coding required!).
Basically you just give Whisper some examples of how you want the text to look like.

Note that my use case is very different and it doesn't involve any email addresses,
so perhaps someone else is able to give you better advice on this specific problem.

**Reply by Rumeysakeskin** [reply]
Thank you @misutoneko ! I will try prompting.

**Comment by misutoneko** [comment]
In addition to the options listed above, there's also the _obvious_ have-the-compute-dont-care answer...

As you perhaps have realized by now, the problem with prompting is that Whisper doesn't really know the context at the same level as you.
But LLMs are supposed to be experts in languages, so you could try feeding the resulting text to an LLM and ask it to fix it for you.

Actually I don't like this method very much (seems excessive), but who cares if it gets the job done.

**Comment by gongouveia** [comment]
@Rumeysakeskin Hello, you can use my tool https://github.com/gongouveia/Whisper-Synthetic-ASR-Dataset-Generator to simply fine tune the model with a few dozen examples, you can record some samples of audio saying emails, translate it with whisper and edit the answers. In the end you can just fine tune whisper with this edited answers.
Hope it can help, my toll is really for this kind of cases
